tests/tests/test_state_scripts.py

```diff
@@ -632,6 +764,139 @@ class TestStateScripts(MenderTesting):
+            return
+
+        client = env.host_string
+        work_dir = "test_state_scripts.%s" % client
+
+        script_content = '#!/bin/sh\n\necho "$(basename $0)" >> /data/test_state_scripts.log\n'
+        # Even though this script is actually run twice, only show it once in the logs
+        script_stop_mender_on_entry = '#!/bin/sh\ntest "$(grep -c "$(basename $0)" /data/test_state_scripts.log)" -eq 0 && echo "$(basename $0)" >> /data/test_state_scripts.log && systemctl stop mender\nexit 0\n'
+        script_failure_content = script_content + 'sync\necho b > /proc/sysrq-trigger\n' # flush to disk before killing
+        script_reboot_once = '#!/bin/sh\ntest "$(grep -c "$(basename $0)" /data/test_state_scripts.log)" -eq 0 && echo "$(basename $0)" >> /data/test_state_scripts.log && sync && echo b > /proc/sysrq-trigger\necho "$(basename $0)" >> /data/test_state_scripts.log\nexit 0'
+        script_error_content = script_content + "exit 1"
+        broken_image = test_set.get("Rollback", False)
+
+        # Put artifact-scripts in the artifact.
+        artifact_script_dir = os.path.join(work_dir, "artifact-scripts")
+
+        if os.path.exists(work_dir):
+            shutil.rmtree(work_dir, ignore_errors=True)
+
+        os.mkdir(work_dir)
+        os.mkdir(artifact_script_dir)
+
+        new_rootfs = os.path.join(work_dir, "rootfs.ext4")
+        shutil.copy(conftest.get_valid_image(), new_rootfs)
+
+        ps = subprocess.Popen(
+            ["debugfs", "-w", new_rootfs], stdin=subprocess.PIPE)
+        ps.stdin.write("cd /etc/mender\n" "mkdir scripts\n" "cd scripts\n")
+        ps.stdin.close()
+        ps.wait()
+
+        for script in test_set.get("ScriptOrder"):
+            if not script.startswith("Artifact"):
+                # Not an artifact script, skip this one.
+                continue
+            with open(os.path.join(artifact_script_dir, script), "w") as fd:
+                if script in test_set.get("RebootScripts", []):
+                    fd.write(script_failure_content)
+                if script in test_set.get("RebootOnceScripts", []):
+                    fd.write(script_reboot_once)
+                elif script in test_set.get("StopSecondPartition", []):
+                    fd.write(script_stop_mender_on_entry)
+                elif script in test_set.get("ErrorScripts", []):
+                    fd.write(script_error_content)
+                else:
+                    fd.write(script_content)
+
+        # Now create the artifact, and make the deployment.
+        device_id = Helpers.ip_to_device_id_map([client])[client]
+        deployment_id = common_update_procedure(
+            install_image=new_rootfs,
+            broken_image=broken_image,
+            verify_status=True,
+            devices=[device_id],
+            scripts=[artifact_script_dir])[0]
+
+        try:
+
+            orig_part = Helpers.get_active_partition()
+
+            token = Helpers.place_reboot_token()
```

Comment:
This has to be placed before `common_update_procedure` is called, otherwise there is a race condition.

Reply:
I take it this no longer applies with the new implementation?

Reply:
The fact that it needs to be before `common_update_procedure` still applies. Translated to the new reboot detector this means that `common_update_procedure` should be inside the `with` scope.